src/client/commands/wait.rs

```diff
@@ -0,0 +1,75 @@
+use crate::{rpc_call, JobId};
+
+use crate::client::job::job_status;
+use crate::client::status::{is_terminated, Status};
+use crate::transfer::connection::ClientConnection;
+use crate::transfer::messages::{FromClientMessage, JobInfoRequest, JobSelector, ToClientMessage};
+use std::thread::sleep;
+use std::time::Duration;
+
+pub async fn wait_on_job(
+    mut connection: ClientConnection,
+    selector: JobSelector,
+) -> anyhow::Result<()> {
+    let connection_ref = &mut connection;
+    let response = rpc_call!(
+        connection_ref,
+        FromClientMessage::JobInfo(JobInfoRequest {
+            selector,
+        }),
+        ToClientMessage::JobInfoResponse(r) => r
+    )
+    .await?;
+    let job_ids: Vec<JobId> = response
+        .jobs
+        .iter()
+        .filter(|job_info| {
+            vec![Status::Running, Status::Waiting]
+                .iter()
+                .any(|status| status == &job_status(&job_info))
+        })
+        .map(|info| info.id)
+        .collect();
+
+    if job_ids.is_empty() {
+        log::warn!("There are no jobs to wait for");
+    } else {
+        let mut current_jobs = 0;
+        let total_jobs = job_ids.len();
+
+        for job_id in job_ids {
```

Comment:
Waiting one by one could in theory generate a lot of requests to the server, if there is a large number of jobs (i.e. million jobs = million requests).
@spirali is this OK? Or should the code instead request the status of a set of tasks at once?

Reply:
Yes you are right, we should expect that user may wait on relatively a large number of jobs

Reply:
That one was for a single use-case, I would rather not reinvent the wheel. But it's your call.